### DISCUSSION TITLE
Add a more practical way to use shaders, filters

### DISCUSSION THREAD

**Discussion by youtubeeicarlos** [mamedev/discussions#129] [Ideas]
Add a more practical way to use shaders, filters.

It would be nice to have an option to use a shader/filter in all games so there is no need to activate them one by one.

As well as a more practical way to use shaders, custom filters, having to edit .ini causes a huge headache, especially when the file gives an error, some games using console cards could gain support for the same shaders and filters that are on that console, for example, Zinc is a PS1 card, you could use what's on the Duckstation to render games made for the Zinc arcade, an FXAA or MSAA in mame would be very useful

**Comment by sairuk** [comment]
In my opinion the ini files provide more flexibility than ui elements with some basic knowledge of the config structure in mame. I do agree the ui could probably provide some generalised chain selection since I haven't noted any ui option for global chain selection etc but i barely use the ui so may have missed it, I'll share a little bit of why/how this works out for me, maybe you can take something from it or perhaps folks can suggest some better options. 

refer: https://docs.mamedev.org/advanced/multiconfig.html for info on how the config file structure works

for example to configure (include only the settings you want changed)
- global shaders configure `mame.ini`
- system/bios shaders for create a `<systemname>.ini` or `<biosname>.ini`
- shaders for all systems attached to a raster screen create a `raster.ini`
- shaders all horizontal games create a `horizontal.ini`

the above goes for all other settings in the `mame.ini`, they can be overridden in additional configs and there is more flexibility in the config structure refer docs

mame has multiple screen types and it makes little sense in my opinion to run the crt-geom-deluxe chain against systems with lcd screens so I prefer to target raster screens over a global `mame.ini` change.
```
$ ./mame -lx | xmllint --xpath '//display/@type' - | sort -u
 type="lcd"
 type="raster"
 type="svg"
 type="vector"
```

as an example my local config right now (i use either a frontend, cli  or very occasionally mame ui) breaks down like this

`mame.ini` -> set rompath and inipath only leave all defaults (for now)
`raster.ini` -> enable bgfx and crt-geom-deluxe
```
video                      bgfx
bgfx_backend               vulkan
bgfx_screen_chains         crt-geom-deluxe
```
`neogeo.ini` -> to enable a custom bios
```
bios          unibios40
```

**Reply by youtubeeicarlos** [reply]
For ordinary people this is hellish, not everyone has time to study how the emulator structure works, sometimes we just want to play a good game with good graphics (or at least something decent), It took me a bloody 6 hours to find how to put shaders in MAME via BGFX and I only found old shader options

**Reply by youtubeeicarlos** [reply]
Imagine having to configure the .ini with each MAME update, there could be a practical interface for that, like even the SNES9X has this, even older emulators like ZNES, Kega Fusion, Mesen...

**Comment by youtubeeicarlos** [comment]
For example, in addition to the standard MAME there is MAMEUI which allows very simplified editing on which shader to use, this could be in MAME (I only use MAMEUI for some games, the "loading" What it does the first time it starts is the only thing that stops me from using it by default, it takes a very long time to load) in it I select the system bios and from there I choose which shader will be, so it is activated for all games on that system
